### ISSUE TITLE
Create Service Bus topic Subscription without redirect

### ISSUE THREAD

**Issue by abuayub** [Azure/ms-rest-browserauth#27]
The following code redirects me for auth. We need to accomplish this without a redirect.
````
import { ServiceBusClient } from '@azure/service-bus';
import { ServiceBusManagementClient, ServiceBusManagementModels, ServiceBusManagementMappers, Subscriptions } from '@azure/arm-servicebus';

const msRestNodeAuth = require("@azure/ms-rest-browserauth");
const clientId = "************************";
const secret = "******************";
const tenantId = "*************";

const authManager = new msRestNodeAuth.AuthManager({
        clientId: clientId,
        tenant: tenantId,
        //redirectUri: "http://localhost:3000/"
      });
      authManager.finalizeLogin().then((authres) => {
          console.log(authres)
          if (!authres.isLoggedIn) {
            // may cause redirects
            authManager.login();
          }
          console.log(authres.creds)
          alert()
        const serviceBusClient = new ServiceBusManagementClient(authres.creds, "*****************");
        const subscriptionClient = new Subscriptions(serviceBusClient);
        return subscriptionClient.createOrUpdate("********", "***********", 
                             process.env.REACT_APP_AZSB_TOPIC_CARDSWIPE, subscriptionId, {
            maxDeliveryCount: 1,
            defaultMessageTimeToLive: 'P0Y0M0DT0H0M5S',
            lockDuration: 'P0Y0M0DT0H0M10S',
            autoDeleteOnIdle: 'P0Y0M0DT0H5M0S'
        })
    }).then((subscription) => {
        console.log(subscription)
        return sbClient.createSubscriptionClient(process.env.REACT_APP_AZSB_TOPIC_CARDSWIPE, subscription.name);
    })
````

**Comment by jonathandturner** [comment]
We investigated this and found that the underlying ADAL library does support more direct authentication that wouldn't require a redirect, but we don't pass through all the settings it would need to perform this. We looked into why this configuration isn't being passed (you can see the full list here: https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Config-authentication-context).

To fix this, this library would need to be updated to take advantage of the additional settings, passing them through the adal authentication context via the constructor. At this point, it should be possible to choose another method, like opening a pop-up to authenticate.

**Comment by HarshaNalluru** [comment]
Hey @abuayub,

We have added a `popUp`(boolean) option to be passed to the AuthManager. This would enable login through a popup window instead of requiring redirects.
The latest version `0.1.5` of the [`@azure/ms-rest-browserauth` library](https://www.npmjs.com/package/@azure/ms-rest-browserauth/v/0.1.5) has just been released with the above feature. We are hoping this would fix your issue.

Here is a new react-app sample [authentication-with-popup](https://github.com/Azure/ms-rest-browserauth/tree/master/samples/authentication-with-popup).

Please check out the latest version and let us know if that fixes your problem.

**Comment by daviwil** [comment]
Fixed by `@azure/ms-rest-browserauth@0.1.5`
